packages/property-grid/src/components/functional/FunctionalElementList.tsx

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import "../ElementList.scss";
+
+import { PresentationLabelsProvider } from "@bentley/presentation-components";
+import {
+  RowItem,
+  SelectionMode,
+  SimpleTableDataProvider,
+  Table,
+} from "@bentley/ui-components";
+import { Icon } from "@bentley/ui-core";
+import classnames from "classnames";
+import * as React from "react";
+
+import { PropertyGridManager } from "../../PropertyGridManager";
+import { createDataProvider, ElementListProps } from "../ElementList";
+
+export const FunctionalElementList = ({
+  iModelConnection,
+  instanceKeys,
+  onBack,
+  onSelect,
+  rootClassName,
+}: ElementListProps) => {
+  const [dataProvider, setDataProvider] =
+    React.useState<SimpleTableDataProvider>();
+
+  const labelsProvider: PresentationLabelsProvider = React.useMemo(() => {
+    return new PresentationLabelsProvider({
+      imodel: iModelConnection,
+    });
+  }, [iModelConnection]);
+
+  React.useEffect(() => {
+    const onMount = async () => {
+      const dp = await createDataProvider(labelsProvider, instanceKeys);
+      setDataProvider(dp);
+    };
+
+    onMount().catch(console.log);
+  }, [labelsProvider, instanceKeys]);
+
+  /** On element selected in table, call the onSelect prop */
+  const onRowsSelected = React.useCallback(
+    async (rowIterator: AsyncIterableIterator<RowItem>, _replace: boolean) => {
+      for await (const row of rowIterator) {
+        if (row.extendedData?.key) {
+          onSelect(row.extendedData.key);
+          return false;
+        }
+      }
+      return false;
+    },
+    [onSelect]
```

Comment:
should we add a note that this prop should be a memoized function? Otherwise this function could be updated unnecessarily with a function as a dependency I think